File: frontend/src/types/product.ts
export interface Producto {
  id: number;
  codigo: string;
  nombre: string;
  descripcion: string | null;
  categoria: string;
  precio: number | string;
  stock: number;
  stock_minimo: number;
  unidad_medida: string;
  is_active: boolean;
  created_at: string;
  updated_at: string;
}

export interface ProductoCreate {
  codigo: string;
  nombre: string;
  descripcion?: string | null;
  categoria: string;
  precio: number;
  stock: number;
  stock_minimo: number;
  unidad_medida: string;
  is_active?: boolean;
}

export interface ProductoUpdate {
  codigo?: string;
  nombre?: string;
  descripcion?: string | null;
  categoria?: string;
  precio?: number;
  stock?: number;
  stock_minimo?: number;
  unidad_medida?: string;
  is_active?: boolean;
}

export interface ProductListResponse {
  items: Producto[];
  total: number;
  skip: number;
  limit: number;
}

export interface ProductAlertItem {
  id: number;
  codigo: string;
  nombre: string;
  categoria: string;
  stock: number;
  stock_minimo: number;
  unidad_medida: string;
}

export interface ProductStats {
  productos_activos: number;
  stock_bajo_count: number;
  categorias_count: number;
  valor_inventario: number | string;
  alertas_stock: ProductAlertItem[];
}

export interface ProductFiltersState {
  search: string;
  categoria: string;
  nivel_stock: string;
  estado: string;
  page: number;
  pageSize: number;
}

File: backend/app/schemas/product.py
from datetime import datetime
from decimal import Decimal
from typing import List, Optional
from pydantic import BaseModel, ConfigDict, Field


class ProductBase(BaseModel):
    codigo: str = Field(..., min_length=1, max_length=50, description="Código único del producto")
    nombre: str = Field(..., min_length=1, max_length=255, description="Nombre del producto")
    descripcion: Optional[str] = Field(None, description="Descripción o presentación del producto")
    categoria: str = Field(..., min_length=1, max_length=100, description="Categoría a la que pertenece")
    precio: Decimal = Field(..., ge=0, description="Precio de venta unitario en Soles")
    stock: int = Field(0, ge=0, description="Cantidad disponible en inventario")
    stock_minimo: int = Field(10, ge=0, description="Nivel mínimo requerido para alertas")
    unidad_medida: str = Field("und", max_length=50, description="Unidad de medida (und, galón, rollo, etc.)")
    is_active: bool = Field(True, description="Estado de disponibilidad")


class ProductCreate(ProductBase):
    pass


class ProductUpdate(BaseModel):
    codigo: Optional[str] = Field(None, min_length=1, max_length=50)
    nombre: Optional[str] = Field(None, min_length=1, max_length=255)
    descripcion: Optional[str] = None
    categoria: Optional[str] = Field(None, min_length=1, max_length=100)
    precio: Optional[Decimal] = Field(None, ge=0)
    stock: Optional[int] = Field(None, ge=0)
    stock_minimo: Optional[int] = Field(None, ge=0)
    unidad_medida: Optional[str] = Field(None, max_length=50)
    is_active: Optional[bool] = None


class ProductResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    codigo: str
    nombre: str
    descripcion: Optional[str] = None
    categoria: str
    precio: Decimal
    stock: int
    stock_minimo: int
    unidad_medida: str
    is_active: bool
    created_at: datetime
    updated_at: datetime


class ProductListResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    items: List[ProductResponse]
    total: int
    skip: int
    limit: int


class ProductAlertItem(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    codigo: str
    nombre: str
    categoria: str
    stock: int
    stock_minimo: int
    unidad_medida: str


class ProductStatsResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    productos_activos: int
    stock_bajo_count: int
    categorias_count: int
    valor_inventario: Decimal
    alertas_stock: List[ProductAlertItem]
